### ISSUE TITLE
Laravel Storage delete() method doesn't return false when the file doesn't exist

### ISSUE THREAD

**Issue by jjdevzinho** [laravel/framework#48634]
### Laravel Version

9.52.15 

### PHP Version

8.2

### Description

The Laravel Storage delete() method doesn't return false when the file doesn't exist or can't be deleted. This is a bug, as it can lead to undetected errors.

Example code
php
$file = "/images/original/{$folder}/{$imageId}.webp";

// The file doesn't exist
Storage::disk('sftp')->delete($file);

// The delete() method returns true, even though the file doesn't exist

### Steps To Reproduce

// The file exists, but the user doesn't have permission to delete it
$file = "/images/original/{$folder}/{$imageId}.webp";

// The `delete()` method returns `true`, even though the file can't be deleted

if (Storage::disk('sftp')->exists($file)) {
    // The file exists, try to delete it
    Storage::disk('sftp')->delete($file);
} else {
    // The file doesn't exist, ignore
}

// The `delete()` method always returns `true`, even if the file can't be deleted

**Comment by crynobone** [comment]
Hey there,

While this may be a legitimate issue, can you first try posting your problem or question on one of the support channels below? If this issue can be definitively identified as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.

* [Laracasts Forums](https://laracasts.com/discuss)
* [Laravel.io Forums](https://laravel.io/forum)
* [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
* [Discord](https://discordapp.com/invite/KxwQuKb)
* [Larachat](https://larachat.co/)
* [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)

Thanks!

**Comment by jjdevzinho** [comment]
Hi,

Thank you for your response.

As you mentioned, I have already opened a post on the two forums you recommended. On Laracasts, I received a response that did not make it clear to me whether the problem is with Laravel or another library that Laravel does not have access to. On Discord, I am still waiting for a response.

I understand that you receive a large volume of issues and problems, but I believe that this issue is important enough to warrant further investigation.

The Laravel Storage delete() method is a fundamental method for file storage. If it does not return false when the file does not exist, this could lead to undetected errors.

I have already provided an example of code that demonstrates the problem. I am also willing to provide more information or help with the investigation of the problem.

I believe it would be beneficial for the Laravel community if this issue were resolved.

Links:
[Discord](https://discord.com/channels/297040613688475649/1160627080824885340)
[Laracasts](https://laracasts.com/discuss/channels/laravel/laravel-storage-delete-method-doesnt-return-false-when-the-file-doesnt-exist)

**Comment by jjdevzinho** [comment]
> Hey there,
> 
> While this may be a legitimate issue, can you first try posting your problem or question on one of the support channels below? If this issue can be definitively identified as a bug, feel free to open up a new issue with a link to the original one and we'll gladly help you out.
> 
> * [Laracasts Forums](https://laracasts.com/discuss)
> * [Laravel.io Forums](https://laravel.io/forum)
> * [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
> * [Discord](https://discordapp.com/invite/KxwQuKb)
> * [Larachat](https://larachat.co/)
> * [IRC](https://web.libera.chat/?nick=laravelnewbie&channels=#laravel)
> 
> Thanks!

Hi,

Thank you for your response.

As you mentioned, I have already opened a post on the two forums you recommended. On Laracasts, I received a response that did not make it clear to me whether the problem is with Laravel or another library that Laravel does not have access to. On Discord, I am still waiting for a response.

I understand that you receive a large volume of issues and problems, but I believe that this issue is important enough to warrant further investigation.

The Laravel Storage delete() method is a fundamental method for file storage. If it does not return false when the file does not exist, this could lead to undetected errors.

I have already provided an example of code that demonstrates the problem. I am also willing to provide more information or help with the investigation of the problem.

I believe it would be beneficial for the Laravel community if this issue were resolved.

Links:
[Discord](https://discord.com/channels/297040613688475649/1160627080824885340)
[Laracasts](https://laracasts.com/discuss/channels/laravel/laravel-storage-delete-method-doesnt-return-false-when-the-file-doesnt-exist)

**Comment by crynobone** [comment]
https://laracasts.com/discuss/channels/laravel/laravel-storage-delete-method-doesnt-return-false-when-the-file-doesnt-exist?page=1&replyId=908392

The reply here is clear, SFTP doesn't return/throw error if there's no file to delete and this is outside of Laravel Framework source code.

**Comment by jjdevzinho** [comment]
> https://laracasts.com/discuss/channels/laravel/laravel-storage-delete-method-doesnt-return-false-when-the-file-doesnt-exist?page=1&replyId=908392
> 
> The reply here is clear, SFTP doesn't return/throw error if there's no file to delete and this is outside of Laravel Framework source code.

Thank you for your time and attention to my issue.

I would like to report that I also noticed the problem using Storage::disk('s3')->delete(). The code I used to demonstrate the problem is as follows:

$file = "/images/original/{$folder}/{$imageId}.webp";
// The file does not exist
Storage::disk('s3')->delete($file);

// The delete() method returns true, even if the file does not exist


As the problem also occurs using Amazon S3, which uses Laravel Storage, I believe the problem is related to Laravel Storage. However, I am not sure if the problem is in Laravel Storage or in an external library that it uses.

As I do not know how to proceed and detect who I should report the problem to, I ask that more experienced people look into this and report the problem in the right place.

I believe this problem is important enough to warrant further investigation.
